### PR TITLE
Rebuild 1.80 with newer compilers & enable 1.78 migrations

### LINKED ISSUE #128: @conda-forge-admin, please re-render
@conda-forge-admin, please re-render.

### LINKED ISSUE #169: Possible macOS ABI breakage in boost-python when boost-python is compiled with clang 13 and downstream packages with clang 14?
### Comment:

Unfortunatly, I am not sure about this, but I tought opening an issue anyhow was useful to spread what we are seeing in some feedstocks. 

In two different feedstocks, we experienced crashes with `63893 Illegal instruction: 4` error on importing the python module when due to re-rendering the downstream packages were compiled with clang 14, while everything was working fine with clang 13:
* https://github.com/conda-forge/eigenpy-feedstock/pull/94#discussion_r966767929
* https://github.com/conda-forge/pinocchio-feedstock/pull/75#issuecomment-1237435999

In one of these cases, for debugging we run the complete C++ testsuite compiled with clang 14 (https://github.com/conda-forge/pinocchio-feedstock/pull/76), and everything works fine, so we think that the problem may be related to boost-python. At the moment we did not tried to recompile boost-python with clang 14 to see if it solves the problem, but one of the hypothesis is that for some reason the ABI of boost-python changes between clang 13 and clang 14. 

I am aware that this is not a complete bug report, but I prefered anyhow to have a place were to track this and report what we find.

### PR DESCRIPTION
**Pull request by conda-forge-linter**
Hi! This is the friendly automated conda-forge-webservice.

I've rerendered the recipe as instructed in conda-forge/boost-cpp-feedstock#128.

Here's a checklist to do before merging.
- [x] Bump the build number if needed.

Fixes https://github.com/conda-forge/boost-feedstock/issues/169

### CODE REVIEW

File: conda-forge.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 bot:
   abi_migration_branches:
-  - 1.72.x
-  - 1.74.x
+  - 1.78.x
```
Comment (jakirkham) [inline conda-forge.yml:3]: xref: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3599

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Review by isuruf — CHANGES_REQUESTED** [review]
Same as https://github.com/conda-forge/boost-cpp-feedstock/pull/130#issuecomment-1297532047

**Comment by jakirkham** [comment]
We need to at least include the updated migration branches. Can you please push that in a way that you are comfortable with?

**Comment by h-vetinari** [comment]
We now have 1.82 and will migrate to that soon. All other changes need to happen on https://github.com/conda-forge/boost-feedstock now.

**Pull request closed without merging**